src/checkers/inference/dataflow/InferenceTransfer.java

```diff
@@ -401,4 +408,94 @@ private InferenceAnalysis getInferenceAnalysis() {
     private boolean isDeclarationWithInitializer(AssignmentNode assignmentNode) {
         return (assignmentNode.getTree().getKind() == Tree.Kind.VARIABLE);
     }
+
+    private void createComparisonVariableSlot(Node node, CFStore thenStore, CFStore elseStore) {
+        // Only create refinement comparison slot for variables
+        // TODO: deal with comparison between more complex expressions
+        Node var = node;
+        if (node instanceof AssignmentNode) {
+            AssignmentNode a = (AssignmentNode) node;
+            var = a.getTarget();
+        }
+        if (!(var instanceof LocalVariableNode) && !(var instanceof FieldAccessNode)) {
+            return;
+        }
+        Tree tree = var.getTree();
+        ConstraintManager constraintManager = InferenceMain.getInstance().getConstraintManager();
+        SlotManager slotManager = getInferenceAnalysis().getSlotManager();
+
+        AnnotatedTypeMirror atm = typeFactory.getAnnotatedType(tree);
```

Comment:
Shouldn't this use `getAnnotatedTypeLhs`?

Reply:
We have a problem in getting the declared type using `getAnnotatedTypeLhs` in CFI, because of the mutable ATM in the `elementToAtm` cache (I have an open PR https://github.com/opprop/checker-framework-inference/pull/362 intended to fix it by duplicating the ATM before writing or after reading). 

The effect is, `getAnnotatedTypeLhs` give us the refined type right after initialization, no matter how many times the variable is reassigned.

A common workaround for this issue in CFI is to use `getAnnotatedType` (which is usually a `RefinementVariableSlot`), and then get the declared type of the `RefinementVariableSlot`, like https://github.com/opprop/checker-framework-inference/blob/8a29a0e4d2349bf108ea0141407925ea27b0911a/src/checkers/inference/dataflow/InferenceTransfer.java#L222
and